snapcraft/file_utils.py

```diff
@@ -414,3 +414,23 @@ def get_resolved_relative_path(relative_path: str, base_directory: str) -> str:
     filename_relpath = os.path.relpath(filename_abspath, base_directory)
 
     return filename_relpath
+
+
+def link_or_copy_files(
+    src_base_dir: str, dst_base_dir: str, file_list: List[str]
```

Comment:
can you make the arguments mandatory by adding a leading *? As I saw the make plugin I noticed I had to come back to this to look at what the 3rd argument was.

How about just using src_dir and dst_dir, they seem pretty standard names already.

Reply:
What if I want to copy ALL files? Should file_list default to None? That might mean os.walk should be used instead.